### PR TITLE
LF-4523: Animals internal identifier changes after data updates

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

Previously, animals/batches' `internal_identifier` was determined solely based on `created_at`. When `created_at` values were identical, animals and batches sometimes received different `internal_identifier` values upon regenerating the view.
This update assigns `internal_identifier` based on `created_at`, `id`, and `batch`, ensuring consistency.

- Add migration to recreate `animal_union_batch_id_view` for stable `internal_identifier`.
- Add a test. (In the test, the view is regenerated 10 times to ensure the test almost always fails with the original view)

The original migration and SQL can be found here ([PR](https://github.com/LiteFarmOrg/LiteFarm/pull/3125)):
- [migration](https://github.com/LiteFarmOrg/LiteFarm/blob/747f76be208ab40fd8bdb42390b0368b583b1fad/packages/api/db/migration/20240207214919_create_animal_union_batch_internal_id_view_modify_animal_batch_tables.js)
- [SQL](https://github.com/LiteFarmOrg/LiteFarm/blob/747f76be208ab40fd8bdb42390b0368b583b1fad/packages/api/db/sql/20240207214919_animal_union_batch_id_view.sql)

Screenshot 2024-11-07 at 5 36 29 PM
Screenshot 2024-11-07 at 5 36 48 PM

Jira link: https://lite-farm.atlassian.net/browse/LF-4523

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [x] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [x] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/api/db/migration/20241108052304_update_animal_union_batch_internal_id_view.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    throw error; // Rethrow the error to ensure the migration fails
+  }
+};
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+export const down = async (knex) => {
+  await knex.schema.dropView('animal_union_batch_id_view');
```
Comment (Duncan-Brain) [inline packages/api/db/migration/20241108052304_update_animal_union_batch_internal_id_view.js:48]: Should this be in try catch too?

**Review by Duncan-Brain — DISMISSED** [review]
just left one comment but I approve either way.

File: packages/api/db/migration/20241108052304_update_animal_union_batch_internal_id_view.js (hunk 1 above)
Comment (SayakaOno) [reply packages/api/db/migration/20241108052304_update_animal_union_batch_internal_id_view.js]: Thank you, I added it!

**Review by Duncan-Brain — APPROVED** [review]

**Review by antsgar — APPROVED** [review]
Thanks for catching this and fixing it!

**Pull request merged**